src/lib/routes/client-api/feature.ts

```diff
@@ -156,13 +199,37 @@ export default class FeatureController extends Controller {
             : await this.resolveFeaturesAndSegments(query);
 
         if (this.clientSpecService.requestSupportsSpec(req, 'segments')) {
-            res.json({ version, features, query, segments });
+            this.openApiService.respondWithValidation(
+                200,
+                res,
+                featureSchema.$id,
+                { version, features, query: { ...query }, segments },
+            );
         } else {
-            res.json({ version, features, query });
+            this.openApiService.respondWithValidation(
+                200,
+                res,
+                featureSchema.$id,
+                { version, features, query: { ...query } },
+            );
         }
     }
 
-    async getFeatureToggle(req: IAuthRequest, res: Response): Promise<void> {
+    async getFeatureToggle(
+        req: IAuthRequest<{ featureName: string }, ClientFeaturesQuerySchema>,
+        res: Response<
+            Omit<
+                FeatureSchema,
```

Comment:
I think we'll need a new schema here, if `FeatureSchema` is this different from what the client API returns. 

Omitting fields will make the type work, but `respondWithValidation` will probably still complain. 

We should fire up a client SDK to if there are any warnings from `respondWithValidation` at least.